src/main/java/com/openlattice/shuttle/dates/JavaDateTimeHelper.java

```diff
@@ -84,62 +126,64 @@ private boolean checkDatePatternIsTwoDigitYear( String datePattern ) {
         return yyMatch && !yyyyMatch;
     }
 
-    public LocalDate parseDateTimeAsDate(String date) {
-        if (shouldIgnoreValue(date))
+    public LocalDate parseDateTimeAsDate( String date ) {
+        if ( shouldIgnoreValue( date ) )
             return null;
-        OffsetDateTime odt = parseDateTime(date);
-        if (odt == null)
+        OffsetDateTime odt = parseDateTime( date );
+        if ( odt == null )
             return null;
         LocalDateTime ldt = odt.toLocalDateTime();
-        if (ldt == null) {
+        if ( ldt == null ) {
             return null;
         }
-        return ldt.atZone(tz.toZoneId()).toLocalDate();
+        return ldt.atZone( tz.orElse( Constants.DEFAULT_TIMEZONE ).toZoneId() ).toLocalDate();
```

Comment:
A cleaner way to do this is to change the `tz` field to not be an optional, allow passing in `Optional<TimeZone>` in the constructor and then on line 28, do a `tz = tz.orElse( Constants.DEFAULT_TIMEZONE )`

Reply:
➕

Reply:
making tz optional and changing logic depending on presence is the goal of this PR.

Reply:
Making the constructor parameter optional doesn't mean that the private class member needs to be optional. It's much cleaner to handle the timezone assignment logic upfront in the constructor, and then reference that value within the rest of the class.

Reply:
Ok looking again, it seems like `tz` is only ever used to be converted to its zoneId. I'd recommend doing that whole conversion upfront, storing `zoneId` as a private class member instead of `tz`, and using a boolean to track whether a timezone was provided (if that's needed) -- this way the implication of `tz` being present/absent is a bit more explicit.